jgrapht-core/src/main/java/org/jgrapht/GraphTests.java

```diff
@@ -154,6 +154,11 @@
     public static <V, E> boolean isWeaklyConnected(Graph<V, E> graph)
     {
         Objects.requireNonNull(graph, GRAPH_CANNOT_BE_NULL);
+
+        if (!graph.getType().isDirected()) {
```

Comment:
Please remove this check. The definition of a weakly connected graph also holds for undirected/mixed graphs. It may not make too much sense to invoke isWeaklyConnected on a undirected graph, since the answer is always 'true', but that's fine.